### ISSUE TITLE
Strongly typed strides

### ISSUE THREAD

**Issue by bluss** [bluss/rust-ndarray#283]
- Separate type for `D: Dimension` and it's corresponding stride
- Strides are signed integers (isize)

**Comment by Robbepop** [comment]
I like this refactoring!
Personally I was slightly confused about this when scanning through the source code of ndarray.

**Comment by bluss** [comment]
It does keep it simple, less generics noise and repetition. However it is time..

**Comment by Robbepop** [comment]
Good decision - 0.9 is already pretty big!

**Comment by bluss** [comment]
Yeah; I don't know how I found so many technically breaking changes to add. There will always be intensive and not so intensive periods.

**Comment by Robbepop** [comment]
Yes, but I personally really like the way you handle refactoring of code and breaking changes in your projects. You do not fear them, you approach them and simply introduce all those new shiny features instead of going with an overly conservative approach that avoids to ever break anything which sadly is a strategy in a few crates ... 👍
But who am I to judge. ;)

**Comment by bluss** [comment]
In this case they have been many technically breaking but almost no actual effect on end users. All changes are judged for their effect, not the technically breaking status. Which is why it is hard to change how dimensions work.
